portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -86,16 +100,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     /* Get the input from the form. */
+    String name = request.getParameter("name-input");
     String comment = request.getParameter("text-input");
+    String imageUrl = getUploadedFileUrl(request, "image");
+
     if (comment != null) {
       Entity commentEntity = new Entity("Comment");
-      comment.replaceAll("(?i)<(/?script[^>]*)>", "&lt;$1&gt;");
-      commentEntity.setProperty("comment", comment);
+      commentEntity.setProperty("name", safeServerEncoding(name));
+      commentEntity.setProperty("comment", safeServerEncoding(comment));
+      commentEntity.setProperty("imageUrl", imageUrl);
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(commentEntity);
     }
     
     response.sendRedirect("/index.html");
   }
+
+  /* Replace script tags. */
+  private String safeServerEncoding(String text) {
+    return text.replaceAll("(?i)<(/?script[^>]*)>", "&lt;$1&gt;");
+  }
+
+  /** Returns a URL that points to the uploaded file, or null if the user didn't upload a file. */
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
```

Comment:
I go with Uncle Bob Martin (https://twitter.com/unclebobmartin/status/870311898545258497): 

> A comment is a failure to express yourself in code. 

Your comment says more then your code does, so either you should check that there is only one key or you don't care. Either way, I think you can reduce the comments you do in your code. Try to use the code to express what you are doing.